flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/streaming/connectors/kafka/table/KafkaDynamicSource.java

```diff
@@ -202,4 +268,169 @@ public int hashCode() {
 
 		return kafkaConsumer;
 	}
+
+	// --------------------------------------------------------------------------------------------
+	// Metadata handling
+	// --------------------------------------------------------------------------------------------
+
+	private enum ReadableMetadata {
+		TOPIC(
+			"topic",
+			DataTypes.STRING().notNull(),
+			record -> StringData.fromString(record.topic())
+		),
+
+		PARTITION(
+			"partition",
+			DataTypes.INT().notNull(),
+			ConsumerRecord::partition
+		),
+
+		HEADERS(
+			"headers",
+			// key and value of the map are nullable to make handling easier in queries
+			DataTypes.MAP(DataTypes.STRING().nullable(), DataTypes.BYTES().nullable()).notNull(),
+			record -> {
+				final Map<StringData, byte[]> map = new HashMap<>();
+				for (Header header : record.headers()) {
+					map.put(StringData.fromString(header.key()), header.value());
+				}
+				return new GenericMapData(map);
+			}
+		),
+
+		LEADER_EPOCH(
+			"leader-epoch",
```

Comment:
True, then I need to change also the recommendations in `SupportsMetadata` interfaces:
```
	 * <p>Metadata key names follow the same pattern as mentioned in {@link Factory}. In case of duplicate
	 * names in format and source keys, format keys shall have higher precedence.
```

Reply:
I would suggest to keep it as it is. Users can use backticks and leader epoch is not very frequently used. Furthermore, once we introduce metadata for formats such as `debezium-json.ingestion-timestamp` it would be confusing if the format identifier changes from `debezium-json` to `debezium_json` for metadata.

Reply:
I see.

Reply:
Is that possible to remove the prefix of `debezium-json`? The Javadoc also says "In case of duplicate names in format and source keys, format keys shall have higher precedence." 

So far, the metadata keys of format and source are very different.

Reply:
Yes, we can shorten them. The idea was to design the metadata similar to regular options. So if key and value are defined they would get a `key.` and `value.` prefix. That should be enough. We don't need the `debezium-json`.

Reply:
But even for `key.` and `value.`, users would need to backticks or the FROM clause. I would stick to the naming convention to not cause confusion.